eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,160 @@
+          WorkingDirectory: $(SpecRepoDirectory)
+        - Name: ${{ parameters.SdkRepoOwner }}/${{ parameters.SdkRepoName }}
+          Commitish: ${{ parameters.SdkRepoBranch }}
+          WorkingDirectory: $(SdkRepoDirectory)
+
+    - script: |
+        if [ "${{ parameters.SpecRepoCommit }}" = "HEAD" ]; then
+          cd $(SpecRepoDirectory)
+          default_commit=$(git rev-parse HEAD)
+          echo "##vso[task.setvariable variable=SpecRepoCommit]$default_commit"
+          echo "SpecRepoCommit variable set to default commit: $default_commit"
+        else
+          echo "##vso[task.setvariable variable=SpecRepoCommit]${{ parameters.SpecRepoCommit }}"
+          echo "SpecRepoCommit variable set to: ${{ parameters.SpecRepoCommit }}"
+        fi
+      displayName: 'Set SpecRepoCommit variable'
+
+    - task: NodeTool@0
+      inputs:
+        versionSpec: '22.x'
+      displayName: 'Install Node.js'
+
+    - script: |
+        npm install -g @azure-tools/spec-gen-sdk
+      displayName: 'Install spec-gen-sdk'
+
+    - script: |
+        optional_params=""
+        sdk_gen_info="sdk generation from Config : "
+
+        if [ "${{ parameters.ConfigType }}" = "TypeSpec" ]; then
+          optional_params="$optional_params --tsp-config-relative-path ${{ parameters.TspConfigPath }}"
+          sdk_gen_info="$sdk_gen_info '${{ parameters.TspConfigPath }}',"
+        else
+          optional_params="$optional_params --readme-relative-path ${{ parameters.ReadmePath }}"
+          sdk_gen_info="$sdk_gen_info '${{ parameters.ReadmePath }}',"
+        fi
+        if [ "${{ parameters.ApiVersion }}" != "not-specified" ]; then
+          optional_params="$optional_params --api-version ${{ parameters.ApiVersion }}"
+          sdk_gen_info="$sdk_gen_info API Version: '${{ parameters.ApiVersion }}',"
+        fi
+
+        sdk_gen_info="$sdk_gen_info and CommitSHA: '$(SpecRepoCommit)'"
+        echo "##vso[task.setvariable variable=GeneratedSDKInformation]$sdk_gen_info"
+        echo "Generated SDK Information : $sdk_gen_info"
+
+        spec-gen-sdk \
+          --scp "$(SpecRepoDirectory)" \
+          --sdp "$(SdkRepoDirectory)" \
+          --wf "$(System.DefaultWorkingDirectory)" \
+          -l "${{ parameters.SdkRepoNameForSpecGen }}" \
+          -c "$(SpecRepoCommit)" \
+          $optional_params
+      displayName: 'Generate SDK'
+
+    - ${{ if eq(parameters.CodePushRequired, true) }}:
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
+        parameters:
+          BaseRepoBranch: sdk-generation-from-$(Build.DefinitionName)_$(Build.BuildId)
+          BaseRepoOwner: azure-sdk
+          CommitMsg: $(GeneratedSDKInformation)
+          TargetRepoOwner: ${{ parameters.SdkRepoOwner }}
+          TargetRepoName: ${{ parameters.SdkRepoName }}
+          PushArgs: "--force"
+          WorkingDirectory: $(SdkRepoDirectory)
+          ScriptDirectory: $(SdkRepoDirectory)/eng/common/scripts
+
+    - task: PowerShell@2
```

Comment:
Are there cases we would create a Pull request without pushing the changes?

Reply:
Do you mean if the head branch will be an existed branch? It is always a brand-new branch which will have the `buildId` in its' name.

Reply:
Dozie, could you add this flag check `eq(parameters.CodePushRequired, true)` to the condition either since PR creation only happens when both flags are set to true?